File: 2020/07/2.go
package main

import (
	"bufio"
	"fmt"
	"io"
	"os"
	"strconv"
	"strings"
)

type valueMap map[string]int
type graph map[string]valueMap

func main() {
	file, _ := os.Open("input")
	defer file.Close()

	bags := parseBags(file)
	fmt.Println(numberOfChildren("shiny gold", bags))
}

func numberOfChildren(parent string, bags graph) int {
	sum := 0
	for bag, count := range bags[parent] {
		sum += count * (1 + numberOfChildren(bag, bags))
	}
	return sum
}

func parseBags(input io.Reader) graph {
	scanner := bufio.NewScanner(input)
	scanner.Split(bufio.ScanLines)

	bags := graph{}
	for scanner.Scan() {
		line := strings.Split(strings.TrimSuffix(scanner.Text(), "."), " bags contain ")

		parent := line[0]
		bags[parent] = valueMap{}

		if line[1] != "no other bags" {
			for _, child := range strings.Split(line[1], ", ") {
				child = strings.TrimSuffix(strings.TrimSuffix(child, "s"), " bag")
				spec := strings.SplitN(child, " ", 2)
				count, _ := strconv.Atoi(spec[0])
				child = spec[1]
				bags[parent][child] = count
			}
		}
	}
	return bags
}

File: 2016/01/2.py
Ps = [(0, 0)]
O = (1, 0)
for i in open("input").read().split(", "):
    if i[0] == 'L':
        O = (O[1], -O[0])
    elif i[0] == 'R':
        O = (-O[1], O[0])
    for _ in range(int(i[1:])):
        P = (Ps[-1][0] + O[0],
             Ps[-1][1] + O[1])
        if P in Ps:
            print(abs(P[0]) + abs(P[1]))
            exit(0)
        Ps.append(P)
